src/functional/ffront/past_to_itir.py

```diff
@@ -112,9 +112,9 @@ def _visit_slice_bound(
         if slice_bound is None:
             lowered_bound = default_value
         elif isinstance(slice_bound, past.Constant):
-            assert (
-                isinstance(slice_bound.type, common_types.ScalarType)
-                and slice_bound.type.kind == common_types.ScalarKind.INT64
+            assert isinstance(slice_bound.type, common_types.ScalarType) and (
+                slice_bound.type.kind == common_types.ScalarKind.INT64
+                or slice_bound.type.kind == common_types.ScalarKind.INT32
```

Comment:
If `slice_bound.type.kind` has always the same value as `common_types.ScalarType.INT`, if should be just checked against `common_types.ScalarType.INT`. Otherwise I think using the `in` operator could be more readable, like:

```python
slice_bound.type.kind in (common_types.ScalarKind.INT32, common_types.ScalarKind.INT64)
```

Reply:
Yeah, that's a good question. Slices should probably be mapped to `ptrdiff_t` (or `intptr_t`)  in the generated C++ or CUDA code regardless of what numpy says. Should we add a `common_types.ScalarType.INTPTR`, or the existing `common_types.ScalarType.DIMENSION` is already for this?

Reply:
I think I'm gonna just simplify it with a tuple and `in` as you suggested, so that it won't fail if somebody uses `numpy.int32` or `int64` to force the type.

Reply:
> so that it won't fail if somebody uses `numpy.int32` or `int64` to force the type.

I think that case can't happen because the values in the PAST are generated from values in a Python AST, which is created directly by the Python parser and it only uses `int`s for constants. It should be safe (and right) to be strict here and allow only `ScalarType.INT`

Reply:
Could you not do something like:
```python
#program({"CONST": np.int32(42)})
def program(...):
  ... 
  op(data[CONST:-CONST])
```
This may not be a literal in the AST, but it's still a constant, right?

Reply:
Alright, let's have it with INT for now.